### ISSUE TITLE
Update nsfodp-maven-plugin to 3.8.1

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#985]
@JesseProminic reported this error while testing Domino On Disk Project with HCL Notes 12:

    : [ERROR] Failed to execute goal org.openntf.maven:nsfodp-maven-plugin:3.4.0:compile (default-compile) on project nsf-moonshine-domino: Exception while compiling the NSF: Encountered exception while launching application: Could not find JVM at /Applications/HCL Notes.app/jre/Contents/Home/bin -> [Help 1]

After some investigation we found that the jvm path had changed in the HCL Notes installation:

    # 11.0
    /Applications/HCL Notes.app/jre/Contents/Home/bin/java
    # 12.0
    /Applications/HCL Notes.app/Contents/jre/Contents/Home/bin/java

I see that there were some other [JVM changes with HCL Notes 12](https://help.hcltechsw.com/domino/beta/12.0.1/wn_new_java_runtime_environment.html), so we should watch for problems with this.

I checked https://github.com/OpenNTF/org.openntf.nsfodp and found that there were newer versions available.  After upgrading from 3.4.0 to 3.8.1, I was able to clear the error with a simulated environment on my local machine:

```
      <plugin>
        <groupId>org.openntf.maven</groupId>
        <artifactId>nsfodp-maven-plugin</artifactId>
        <version>3.8.1</version>
        <extensions>true</extensions>
      </plugin>
```

It looks like the plugin will download a JDK for this update.  This may be a problem for the App Store build, but I'm not sure we will ever get NSFODP working well in the Sandbox anyway:

    : [INFO] Compiling ODP
    : Mar 03, 2022 4:21:20 PM org.openntf.maven.nsfodp.jvm.AbstractMacGitHubJvmEnvironment getJavaHome
    : INFO: Downloading Adoptium Temurin JDK from https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u322-b06/OpenJDK8U-jdk_x64_mac_hotspot_8u322b06.tar.gz

@JesseProminic, please check if this version change resolves the problem for you.  I confirmed that this version still works for me with HCL Notes 11.0 on macOS.  

We will need to update both Domino On Disk Project and Domino Visual Editor.  We will also need to update the version in NSFConverterPortalProject (see https://github.com/Moonshine-IDE/NSFConverterPortalProject/issues/8 )

**Comment by MarkProminic** [comment]
Hello Joel,

I am helping Jesse with this issue, can you confirm the steps you took to upgrade the version?
```
"After upgrading from 3.4.0 to 3.8.1, I was able to clear the error with a simulated environment on my local machine:"
```

**Comment by JoelProminic** [comment]
Sorry @MarkProminic, I see I missed a key piece of information

1. Open/Create a Domino On Disk Project
2. Open `pom.xml` in the project root
3. Search for "nsfodp-maven-plugin".   You should find a section like what I pasted below
4. Change `<version>` from `3.4.0` to `3.8.1`
5. Project > Build with Apache Maven.  This does a clean build already.

```
      <plugin>
        <groupId>org.openntf.maven</groupId>
        <artifactId>nsfodp-maven-plugin</artifactId>
        <version>3.8.1</version>
        <extensions>true</extensions>
      </plugin>
```

Do not update `domino-pom.xml` (I think this is an unnecessary duplicate of pom.xml) or `nsfs/nsf-moonshine/pom.xml` (Does not require updates.

**Comment by JoelProminic** [comment]
I did some testing with @JesseProminic and @MarkProminic, and we saw that the update cleared the JVM error for HCL Notes 12.0 on Mojave and Big Sur.

However, the tests failed after this error with NSD errors.  I think this is probably a separate problem, though, so it is worth updating the plugin version.

**Comment by rat-moonshine** [comment]
'nsfodp-maven-plugin` value updated in `pom.xml` and `domino-pom.xml` file.

**Comment by JoelProminic** [comment]
We found that the correct update location for Domino Visual Editor was:  https://github.com/Moonshine-IDE/Moonshine-IDE/blob/04d9660b915a6d93754376a01dac5a795d9d8c35/ide/MoonshineSharedCore/src/elements/templates/projects/Visual%20Editor%20Project/%24domino-pom.xml.template

I confirmed this is updated now for a new project.

**Comment by JoelProminic** [comment]
Revisiting this on a machine with the JVM error, I found that the build was still using 3.4.0 and failing.  After some more investigation, I found that `nsfs/nsf-moonshine/pom.xml` also has a dependency on `nsfodp-maven-plugin`, and least for Domino Visual Editor.

**Comment by JoelProminic** [comment]
I updated https://github.com/Moonshine-IDE/Moonshine-IDE/blob/b73ecc0cce8d0792fe6ea3b742ab9cf074859b17/ide/MoonshineSharedCore/src/elements/templates/projects/Visual%20Editor%20Project/src_domino_nsfs/nsf-moonshine/pom.xml .   I'll test the update once this is complete.

**Comment by JoelProminic** [comment]
The update worked for me.  I'm still getting NSD errors and other errors on the HostMyApple machine, but I think this is unrelated.
